x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_host_table.ts

```diff
@@ -7,19 +7,28 @@
 
 import { last } from 'lodash';
 import { useMemo } from 'react';
-import type { SnapshotNode, SnapshotNodeMetric } from '../../../../../common/http_api';
-import { HostMetics } from '../components/hosts_table_columns';
+import { SnapshotNode } from '../../../../../common/http_api';
+import { HostMetics, HostNodeRow } from '../components/hosts_table_columns';
 
-type MappedMetrics = Record<keyof HostMetics, SnapshotNodeMetric>;
+export type MappedMetrics = Record<keyof HostMetics, number | null | undefined>;
 
 export const useHostTable = (nodes: SnapshotNode[]) => {
-  const items: MappedMetrics[] = useMemo(() => {
-    return nodes.map(({ metrics, path }) => ({
-      ...last(path),
+  const items: HostNodeRow[] = useMemo(() => {
+    const valuesMapping: Record<keyof HostMetics, 'value' | 'avg' | 'max'> = {
+      cpuCores: 'value',
+      rx: 'avg',
+      tx: 'avg',
+      memory: 'avg',
+      memoryTotal: 'avg',
+    };
+    return nodes.map(({ metrics, path, name }) => ({
+      name,
+      os: last(path)?.os ?? '-',
       ...metrics.reduce((data, metric) => {
-        data[metric.name as keyof HostMetics] = metric;
+        const metricName = metric.name as keyof HostMetics;
+        data[metricName] = metric[valuesMapping[metricName]];
```

Comment:
Should we hand;e null values here too? In case these any of these metrics are null, should we set the default value to `0`?

Reply:
The `NumberCell` [component](https://github.com/elastic/kibana/blob/a59c0482ca87de4a65722f01cc2e713f2862ace1/x-pack/plugins/infra/public/components/infrastructure_node_metrics_tables/shared/components/number_cell.tsx#L18) will handle the `null` case and will display `N/A` in that case - I think it's more accurate when the metric is not present instead of showing `0` - what do you think?